.circleci/config.yml

```diff
@@ -17,12 +17,13 @@ jobs:
       - image: cimg/go:1.16
     steps:
       - checkout
-      - run: curl -L https://git.io/vp6lP | sh # gometalinter
-      - run: PATH=./bin:$PATH gometalinter --deadline=5m --disable-all --enable=gofmt --enable=vet --vendor ./...
       - run: sudo apt-get update
       - run: sudo apt-get install python3-pip
       - run: sudo pip3 install invoke semver pyyaml
       - run: inv checkpatch
+      # The version should match the version used in the linter task in tasks.py
+      - run: curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.39.0
+      - run: golangci-lint run --timeout 5m0s ./...
```

Comment:
It would be great if this could just be `inv lint`.

Then in tasks.py, maybe it should have an option for whether it should run it in `docker` or whether it should run a pre-installed `golangci-lint`?

Similar to other spots in CI config, I just want a single place that ensures we run the same commands for both dev testing and CI testing.

Reply:
That was my plan originally. I had to try it again to remember why I could not do it like that. The reason is that we are running in a docker container. In order to pull a docker image we need to set up a remote docker environment. This remote docker environment does not have access to the files in the primary docker environment. This is documented at: https://circleci.com/docs/2.0/building-docker-images/#mounting-folders. So if you try `inv lint`, you get an error like this 

```
...
e07b05f2a22d: Pull complete
f298da50fa05: Pull complete
5f9f36afbb7f: Pull complete
Digest: sha256:06c0900d94953cfa35df107e36c6f8a3d191f4ce9736ce067eb1862da6e7e7a7
Status: Downloaded newer image for golangci/golangci-lint:v1.39.0
level=error msg="Running error: context loading failed: no go files to analyze"

Exited with code exit status 5
```

I had three options to resolve this:

1. Run the CI stage in a VM (I discounted because it would run slower)
2. Copy the files between containers (I discounted because it would run slower)
3. The current option.

I am happy to change to another option if you prefer as all options weren't great.

Reply:
Yeah i was thinking something like ...

the default behavior:

```
$ inv lint --env container
```

or how we'd run it from the CI config:

```
$ inv lint --env host
```

Reply:
Updated.